src/Integration/src/RabbitMQ/Inbound/RabbitInboundChannelAdapter.cs

```diff
@@ -0,0 +1,263 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.Logging;
+using RabbitMQ.Client;
+using Steeltoe.Common.Contexts;
+using Steeltoe.Common.Retry;
+using Steeltoe.Common.Util;
+using Steeltoe.Integration.Endpoint;
+using Steeltoe.Integration.Rabbit.Support;
+using Steeltoe.Integration.Support;
+using Steeltoe.Messaging;
+using Steeltoe.Messaging.Converter;
+using Steeltoe.Messaging.RabbitMQ;
+using Steeltoe.Messaging.RabbitMQ.Batch;
+using Steeltoe.Messaging.RabbitMQ.Core;
+using Steeltoe.Messaging.RabbitMQ.Extensions;
+using Steeltoe.Messaging.RabbitMQ.Listener;
+using Steeltoe.Messaging.RabbitMQ.Support;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using RabbitConverter = Steeltoe.Messaging.RabbitMQ.Support.Converter;
+
+namespace Steeltoe.Integration.Rabbit.Inbound
+{
+    public class RabbitInboundChannelAdapter : MessageProducerSupportEndpoint // ,IOrderlyShutdownCapable
```

Comment:
Is `IOrderlyShutdownCapable` something that is (or needs to be) documented as a future enhancement?

Reply:
@dtillman do we need this? I only saw this one instance of this interface name.

Reply:
We may not ... I probably added it at one point during the porting effort .. thinking it would be needed .. but obviously sorta changed my mind ;-) ... I would just double check its usage in the Java code and make a determination..

Reply:
The only place it is used is for Jmx bean exporter - I don't think we need it. Removing this.